api/handlers.go

```diff
@@ -123,8 +123,12 @@ func getImagesJSON(c *context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	accepteds, _ := filters["node"]
-	images := []*cluster.Image{}
-
+	// this struct helps grouping images
+	// but still keeps their Engine infos as an array.
+	groupImages := make(map[string]*struct {
+		dockerclient.Image
+		Engines []*cluster.Engine
```

Comment:
@ahmetalpbalkan  It will be serialized to JSON.

In `swarm` there's a way to keep Image/Engine info (the struct inside `cluster/image.go`). This `Engines` here is do the same. It's intended to not discard Image/Engine information.